contract/src/main.rs

```diff
@@ -1364,6 +1379,49 @@ pub extern "C" fn updated_receipts() {
     runtime::ret(CLValue::from_t(updated_receipts).unwrap_or_revert())
 }
 
+#[no_mangle]
+pub extern "C" fn register_owner() {
+    if let ReportingMode::NoReport = utils::get_reporting_mode() {
+        runtime::revert(NFTCoreError::InvalidReportingMode)
+    }
+
```

Comment:
add code comments for and to this entry point explaining its use and helping people understand what the code is doing

Reply:
Aye sir